sql: TestIndexBackfillMergeRetry failed

sql.TestIndexBackfillMergeRetry [failed](https://tanzanite.cluster.engflow.com/invocations/default/cfcd53df-372b-48e1-9de5-9de384b57aa5?testReportRun=1&testReportShard=1&testReportAttempt=1#targets-Ly9wa2cvc3FsOnNxbF90ZXN0) on release-24.1 @ [4aba3c9a58dfce929d406732b28a968db6ea1a18](https://github.com/cockroachdb/cockroach/commits/4aba3c9a58dfce929d406732b28a968db6ea1a18):

Fatal error:

```
panic: test timed out after 39m57s
running tests:
	TestIndexBackfillMergeRetry (38m51s)

```
Stack: 

```
goroutine 230411644 [running]:
testing.(*M).startAlarm.func1()
	GOROOT/src/testing/testing.go:2366 +0x385
created by time.goFunc
	GOROOT/src/time/sleep.go:177 +0x2d
```
<details><summary>Log preceding fatal error</summary>
<p>

```
=== RUN   TestIndexBackfillMergeRetry
    test_log_scope.go:165: test logs captured to: outputs.zip/logTestIndexBackfillMergeRetry3180902934
    test_log_scope.go:76: use -show-logs to present logs inline
```
</p>
</details>

Parameters:
 - <code>attempt=1</code>
 - <code>deadlock=true</code>
 - <code>run=1</code>
 - <code>shard=1</code>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestIndexBackfillMergeRetry.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>


Jira issue: CRDB-43886

--- a/pkg/sql/mvcc_backfiller_test.go
+++ b/pkg/sql/mvcc_backfiller_test.go
@@ -57,7 +57,7 @@ func TestIndexBackfillMergeRetry(t *testing.T) {
 	defer leaktest.AfterTest(t)()
 	defer log.Scope(t).Close(t)
 
-	skip.UnderRace(t, "TODO(ssd) test times out under race")
+	skip.UnderDuress(t, "this test fails under duress")
 
 	params, _ := createTestServerParams()
 

